plugins/modules/foreman_compute_profile.py

```diff
@@ -121,97 +121,49 @@
-    compute_attribute_dict = attributes.copy()
-    for key in ['compute_resource', 'vm_attrs']:
-        if key not in compute_attribute_dict:
-            module.fail_json(msg='compute_attribute must have %s.' % key)
-
-    try:
-        compute_attribute = find_compute_attribute(module, compute_profile_name=compute_profile.name,
-                                                   compute_resource_name=compute_attribute_dict['compute_resource'], failsafe=True)
-        compute_attribute_dict['compute_resource'] = find_compute_resource(module, name=compute_attribute_dict['compute_resource'])
-    except Exception as e:
-        module.fail_json(msg='Failed to find entity: %s ' % e)
-
-    if 'compute_profile' not in compute_attribute_dict:
-        compute_attribute_dict['compute_profile'] = compute_profile
-
-    changed = naildown_entity_state(ComputeAttribute, compute_attribute_dict, compute_attribute, 'present', module)
-
-    return changed
-
-
 def main():
     module = ForemanEntityAnsibleModule(
-        argument_spec=dict(
+        entity_spec=dict(
             name=dict(required=True),
+            compute_attributes=dict(type='nested_list', entity_spec=compute_attribute_entity_spec),
+        ),
+        argument_spec=dict(
             updated_name=dict(),
-            compute_attributes=dict(type='list'),
         ),
     )
 
-    (compute_profile_dict, state) = module.parse_params()
-    name = compute_profile_dict.get('name')
-    updated_name = compute_profile_dict.get('updated_name')
-    compute_attributes = compute_profile_dict.pop('compute_attributes', list())
-
-    if len(compute_attributes) > 0 and state == 'absent':
-        module.fail_json(msg='compute_attributes not allowed when state=absent')
+    entity_dict = module.clean_params()
+    updated_name = entity_dict.get('updated_name')
+    compute_attributes = entity_dict.pop('compute_attributes', None)
 
     module.connect()
 
-    try:
-        # Try to find the compute_profile to work on
-        compute_profile = find_compute_profile(module, name=compute_profile_dict['name'], failsafe=True)
-    except Exception as e:
-        module.fail_json(msg='Failed to find entity: %s ' % e)
-
-    if state == 'present' and updated_name:
-        compute_profile_dict['name'] = updated_name
+    entity = module.find_resource_by_name('compute_profiles', name=entity_dict['name'], failsafe=True)
 
-    compute_profile_dict = sanitize_entity_dict(compute_profile_dict, name_map)
+    if module.state == 'present' and updated_name:
+        entity_dict['name'] = updated_name
 
-    (changed, compute_profile) = naildown_entity(ComputeProfile, compute_profile_dict, compute_profile, state, module)
+    changed, compute_profile = module.ensure_entity('compute_profiles', entity_dict, entity)
 
     # Apply changes on underlying compute attributes only when present
-    if state == 'present':
+    if module.state == 'present' and compute_attributes is not None:
```

Comment:
and then, following https://github.com/theforeman/foreman-ansible-modules/pull/401/files#r314193188, just checking `and compute_attributes` (an empty list is `False`-ish)